File: src/main/java/com/homechange/api/repository/UserRepository.java
package com.homechange.api.repository;

import com.homechange.api.model.User;
import org.springframework.data.jpa.repository.JpaRepository;

/**
 * Created by Nemanja on 5/14/17.
 */
public interface UserRepository extends JpaRepository<User, Long> {

	/**
	 * Method that finds user by username
	 * @param username Username
	 * @return User
	 */
	User findByUsernameIgnoreCase(String username);

	/**
	 * Method that finds user by Email
	 * @param email Email
	 * @return User
	 */
	User findByEmailIgnoreCase(String email);
}

File: src/main/java/com/homechange/api/repository/UserRepository.kt
package com.homechange.api.repository

import com.homechange.api.model.User
import org.springframework.data.jpa.repository.JpaRepository

/**
 * Created by Nemanja on 5/14/17.
 */
interface UserRepository : JpaRepository<User, Long> {

    /**
     * Method that finds user by username
     * @param username Username
     * *
     * @return User
     */
    fun findByUsernameIgnoreCase(username: String?): User?

    /**
     * Method that finds user by Email
     * @param email Email
     * *
     * @return User
     */
    fun findByEmailIgnoreCase(email: String?): User?
}
